mesonbuild/interpreter.py

```diff
@@ -2263,6 +2264,42 @@ def _get_native() -> str:
             else:
                 return _get_native()
 
+    @permittedKwargs({'encoding', 'allow_absolute'})
+    @FeatureNew('meson.get_file_contents', '0.58.0')
+    def get_file_contents_method(
+            self, args: T.Sequence[str], kwargs: dict) -> str:
+        '''
+        Read a file from the source tree and return its value as a decoded
+        string. If the encoding is not specified, the file is assumed to be
+        utf-8 encoded. Paths must be relative by default (to prevent accidents)
+        and are forbidden to be read from the build directory (to prevent build
+        loops)
+        '''
+        if len(args) != 1:
+            raise InterpreterException('expected single positional <path> arg')
+        path = args[0]
+        encoding = kwargs.get('encoding', 'utf-8')
+        allow_absolute = kwargs.get('allow_absolute', False)
+        source_dir = self.interpreter.environment.get_source_dir()
+        build_dir = self.interpreter.environment.get_build_dir()
+        if not allow_absolute and os.path.isabs(path):
+            raise InterpreterException('paths must be relative')
```

Comment:
probably should add `unless the allow_abslolute` keyword is set. I'm not sure that we really need a public keyword for this though.

Reply:
done. I think the `allow_absolute` is a useful escape hatch with defaults that prevent accidents, but I'm happy to remove this logic if you're sure it's not warranted

Reply:
I don't know for sure, but we don't have any other enforcement of absolute/not absolute. @jpakkane?

Reply:
IMHO we should drop allow_absolute, most APIs transparently accept both relative and absolute AFAIK.